sopel/modules/version.py

```diff
@@ -33,22 +33,43 @@ def git_info():
 
 
 @plugin.command('version')
+@plugin.example('.version')
+@plugin.example('.version url')
 @plugin.output_prefix('[version] ')
 def version(bot, trigger):
-    """Display the installed version of Sopel.
+    """Display the installed version of Sopel or a plugin.
 
     Includes the version of Python Sopel is installed on.
     Includes the commit hash if Sopel is installed from source.
     """
-    parts = [
-        'Sopel v%s' % release,
-        'Python: %s' % platform.python_version()
-    ]
-    sha = git_info()
-    if sha:
-        parts.append('Commit: %s' % sha)
-
-    bot.say(' | '.join(parts))
+    plugin = trigger.group(3)
+    if not plugin or plugin.lower() == "sopel":
+        # Sopel version
+        parts = [
+            'Sopel v%s' % release,
+            'Python: %s' % platform.python_version()
+        ]
+        sha = git_info()
+        if sha:
+            parts.append('Commit: %s' % sha)
+
+        bot.say(' | '.join(parts))
+
+    else:
+        # Plugin version
+        if not bot.has_plugin(plugin):
+            bot.say("I don't have a plugin named %r loaded." % plugin)
+            return
+
+        meta = bot.get_plugin_meta(plugin)
+        if meta["source"].startswith("sopel."):
+            version = "v" + release + " (built in)"
+        elif meta["version"] is None:
+            version = "(version unknown)"
+        else:
+            version = "v" + meta["version"]
+
+        bot.say("{} {}".format(plugin, version))
```

Comment:
I like to avoid unnecessary nesting and go for `if-return/do-more` rather than `if/else`. In this case it would allow leaving the original code alone and just inserting the new plugin logic above it.